login page and form.css have been added

hannahlabana opened the pull request.

hannahlabana committed:
login page and form.css have been added

cruzhansm commented on css/login/login-body.css:
```diff
@@ -0,0 +1,73 @@
+.body{
```
![image](https://user-images.githubusercontent.com/58615030/139578375-aab8088a-ca81-41de-ab73-4e02f142024b.png)
![image](https://user-images.githubusercontent.com/58615030/139578383-9df6c71c-acc1-4124-a8e7-0ec139c4d03b.png)

Your alignment of all major elements is wrong (banner, form, and the copyright/footer). Try to view the two images above and compare.
Instead of implementing individual margin-top styling in .body-banner, .body-login, and .body-signup, what you could instead is to add these lines to .body:

33px = vertical distance from header to the top of .body-signup
125px = vertical distance from header to the top of .body-login.
144px = vertical distance from header to the top of .body-banner
144px - 125px = 19px = ~ 20px (difference in heights of .body-banner and .body-login)
144px - 33px = 111px = ~110px (difference in heights of .body-banner and .body-signup)

**padding-top: 33px;** (when signup form is present)
**padding-top: 125px;** (when login form is present)
This will result into .body taking more space than intended.

OR

**margin-top: 33px;** (when signup form is present)
**margin-top: 125px;** (when login form is present)
This will result into .body immediately starting 144px below the header, instead of taking up unnecessary space. I recommend this more, since you could just add .padding-top to .body and the rest if you want to adjust their vertical positioning.

**In either case you choose, you need to modify .body-banner with this:**
**margin-top: 110px** (when signup form is present)
**margin-top: 20px** (when login form is present)
This will adjust the height of .body-banner, so is now in its proper position.

This way, all child elements of .body will immediately start at the correct position below the header. This also means that we have to remove the **margin-top** styling for body-signup, and .body-login.

The end result should look like this:
![image](https://user-images.githubusercontent.com/58615030/139578476-d88256b7-a9b3-4272-9974-7b10671f6392.png)
![image](https://user-images.githubusercontent.com/58615030/139578432-2a4a4f4a-56a8-46db-a238-595ed949c45d.png)

cruzhansm replied:
To see if page matches its design in the prototype, I also suggest doing this:

Open the page in one tab in Inspector tool and in another tab, open the prototype (in Play / Interaction mode; use Ctrl + Shift + I because you can't right click).

If you got the measurements correctly, then when you switch tabs between the two to compare, it should align _perfectly_.

cruzhansm replied:
After doing all of the above, the copyright / footer should also align perfectly with the prototype.

hannahlabana replied:
I was relying on my display instead of using the inspect function which is why I decided to ignore the values on Figma. I've now changed it so that it is exactly like the mock-up.

cruzhansm commented on css/login/login-header.css:
```diff
@@ -0,0 +1,20 @@
+}
+
+.login-header-text{
+  font-weight:800;
```
Useless line, this is overridden by **fw-bold** already. Please remove.

cruzhansm commented on login.html:
```diff
@@ -0,0 +1,127 @@
+      if (password) {
+        login_password.querySelector("input").setAttribute('type','text');
+        login_password.querySelector("i").style.color = '#B9C8D3';
+      } else {
```
Coding convention violation. Please refer to documentation once more.

cruzhansm commented on login.html:
```diff
@@ -0,0 +1,127 @@
+    });
+
+    login_password.querySelector('i').addEventListener('click', () => {
+      if (password) {
```
Do not put spaces between a function and parenthesis, ex: if(password) is the accepted standard.

cruzhansm commented on css/login/login-body.css:
```diff
@@ -0,0 +1,73 @@
+  font-size:18px;
+}
+
+.body-login-signup-link > span, .body-signup-login-link > span{
```
Write multiple long selections in new lines, ex:

.body-login-signup-link > span,
.body-signup-login-link >span { }

cruzhansm commented on css/base/form.css:
```diff
@@ -0,0 +1,69 @@
+}
+
+.form-control, .form-control:focus{
+  width:324px;
```
I think this should not be hard-coded, because there are multiple pages wherein .form-control or inputs are required, such as the Add Listing page, Edit Listing page, etc.

If we were using Tailwindcss, we could've simply done: 

input class=".form-control(width, height)", so that we won't have to bother with anything.

But since we're doing Bootstrap, what we can do instead is to create individual form.css files for each page that has a custom width and height of the inputs.

Hence, all of these properties can stay, with the **exception of: width, height, margin-bottom, margin-left, margin-right,**. 

For each page that requires a custom dimensions, create a form.css file inside a folder with that page's name inside the css folder, and override the .form-control and .form-control:focus classes.

As for this file, since you already have a css/login folder, we can do:

css/login/login-form.css

then override .form-control and .form-control:focus.

cruzhansm commented on css/base/form.css:
```diff
@@ -0,0 +1,69 @@
+}
+
+.form-password{
+  margin-bottom:22px;
```
Since .form-password is really only required in the login/sign up page, I think this can stay intact.

cruzhansm commented on css/base/form.css:
```diff
@@ -0,0 +1,69 @@
+  color:#B9C8D3;
+}
+
+.btn-primary{
```
I've noticed that .btn-primary styling is the same in multiple pages, so add this to components/buttons.css instead.

While the properties that require customization such as width and height shall remain here, and implement the same scheme as discussed above.

cruzhansm commented on css/base/form.css:
```diff
@@ -0,0 +1,69 @@
+  background-color: var(--tph-text-highlight);
+  text-transform:uppercase;
+  font-size: 21px;
+  border-radius:6px;
```
Okay this is honestly super confusing.

There are so many border-radius being implemented in the prototype. **Please** settle for **standard** border-radius for: buttons, content (divs/containers/etc), and anything else that require border-radiuses, and implement them accordingly in the prototype.

cruzhansm commented on css/base/form.css:
```diff
@@ -0,0 +1,69 @@
+  transition:none !important;
+}
+
+.btn-primary:focus{
```
Move to css/buttons.css

cruzhansm commented on css/base/form.css:
```diff
@@ -0,0 +1,69 @@
+
+
+
+.btn-primary:hover{
```
Same as above.

cruzhansm commented on css/base/form.css:
```diff
@@ -0,0 +1,69 @@
+  height:45px;
+  border-radius: 73px;
+  background-color: var(--tph-tertiary);
+  color:#B9C8D3;
```
There are also a lot of colors being used in the prototype that are not hard-coded within our css variables, such as this. Please identify their purpose and create :root variables for them with the appropriate name. Thanks!

hannahlabana replied:
That is actually the hex value for input values, but it doesn't work on mine so I added an --input variable (to resolve this).

cruzhansm reviewed (changes requested):
**CODING CONVENTIONS**
1. Please apply one spacing after the property name for CSS styling.
    Example: 

    .btn-primary {
       code here
     }

    AS OPPOSED TO

    .btn-primary{
      code here
    }    

2. Let your code breath. Same as with the operators in the source code, please, please, please put a space after colons ( : ) when styling properties in CSS.
    Example:

    .btn-primary {
       border: none;
     }

    AS OPPOSED TO

    .btn-primary{
       border:none;
    }

**NAMING CONVENTIONS**
1. Do not mix up snake_case and camelCase. camelCaseStandard is _strictly_ for HTML, CSS, and JavaScript, while snake_case_standard is _stricly_ for PHP files. Rename your variables.

**COMMENTS**
1. Read the attached comments in each file that has a comment on them. If you do not agree with the suggestion and/or proposed solution, feel free to provide an argument for it and inform me.
2. Please move login.html into the src folder. Only index.html stays in the root folder. Delete the .keep file (if it still exists).
3. Aside from all of the above-mentioned, great job on the page! 
4. Once you're done implementing the changes (git push -f origin HEAD, so that it'll reflect in the PR), click "Re-request review" located here:
![image](https://user-images.githubusercontent.com/58615030/139580314-617d2493-d9e3-4404-837b-bae14ac796d5.png)

hannahlabana committed:
revisions for login and form.css

cruzhansm committed:
Update login-body.css

Removed extraneous measurement "px" in 0px styling.

cruzhansm committed:
Update buttons.css

Reordered some classes.
Transferred properties from .btn-primary to .btn, since they apply to all .btn variations.

cruzhansm committed:
Update border-radius.css

Fixed indentation.

cruzhansm reviewed (approved):
Great! Don't worry about refactoring this for now, we'll talk about that later.

cruzhansm committed:
Merge branch 'master' into login
